crates/database/src/fee_policies.rs

```diff
@@ -0,0 +1,170 @@
+use {
+    crate::{auction::AuctionId, OrderUid, PgTransaction},
+    bigdecimal::BigDecimal,
+    sqlx::PgConnection,
+    std::ops::DerefMut,
+};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FeePolicy {
+    pub auction_id: AuctionId,
+    pub order_uid: OrderUid,
+    pub kind: FeePolicyKind,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum FeePolicyKind {
+    PriceImprovement {
+        price_improvement_factor: f64,
+        max_volume_factor: Option<f64>,
+        max_absolute_fee: Option<BigDecimal>,
+    },
+    Volume {
+        factor: f64,
+    },
+}
+
+#[derive(Debug, Clone, PartialEq, sqlx::FromRow)]
+struct FeePolicyRow {
+    auction_id: AuctionId,
+    order_uid: OrderUid,
+    price_improvement_factor: Option<f64>,
+    volume_factor: Option<f64>,
+    absolute_fee: Option<BigDecimal>,
+}
+
+pub async fn insert(ex: &mut PgTransaction<'_>, fee_policy: FeePolicy) -> Result<(), sqlx::Error> {
```

Comment:
My reasoning to define it here is so that it could be reusable (potentially) by both orderbook and autopilot. Autopilot and ordebook should probably have their own dto's for this structure.

Reply:
Kept the "user friendly" common type because it encapsulates how the fields are populated for different policy types (IMO autopilot and orderbook should not worry about it).